morphia/src/main/java/org/mongodb/morphia/IndexHelper.java

```diff
@@ -0,0 +1,492 @@
+import org.mongodb.morphia.annotations.Text;
+import org.mongodb.morphia.logging.Logger;
+import org.mongodb.morphia.logging.MorphiaLoggerFactory;
+import org.mongodb.morphia.mapping.MappedClass;
+import org.mongodb.morphia.mapping.MappedField;
+import org.mongodb.morphia.mapping.Mapper;
+import org.mongodb.morphia.mapping.MappingException;
+import org.mongodb.morphia.utils.IndexType;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+import static java.lang.reflect.Proxy.newProxyInstance;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.mongodb.morphia.utils.IndexType.fromValue;
+
+class IndexHelper {
+    private static final Logger LOG = MorphiaLoggerFactory.get(IndexHelper.class);
+    private static final EncoderContext ENCODER_CONTEXT = EncoderContext.builder().build();
+
+    private final AdvancedDatastore datastore;
+    private final Mapper mapper;
+    private final MongoDatabase database;
+
+    IndexHelper(final AdvancedDatastore datastore, final Mapper mapper, final MongoDatabase database) {
+        this.datastore = datastore;
+        this.mapper = mapper;
+        this.database = database;
+    }
+
+    @SuppressWarnings("deprecation")
+    protected com.mongodb.client.model.IndexOptions convert(final IndexOptions options, final boolean background) {
+        if (options.dropDups()) {
+            throw new MappingException("dropDups value has been desupported by the server.  Please set this value to false and "
+                                           + "validate your system behaves as expected.");
+        }
+        com.mongodb.client.model.IndexOptions indexOptions = new com.mongodb.client.model.IndexOptions()
+            .background(options.background() || background)
+            .sparse(options.sparse())
+            .unique(options.unique());
+
+        if (!options.language().equals("")) {
+            indexOptions.defaultLanguage(options.language());
+        }
+        if (!options.languageOverride().equals("")) {
+            indexOptions.languageOverride(options.languageOverride());
+        }
+        if (!options.name().equals("")) {
+            indexOptions.name(options.name());
+        }
+        if (options.expireAfterSeconds() != -1) {
+            indexOptions.expireAfter((long) options.expireAfterSeconds(), TimeUnit.SECONDS);
+        }
+        if (!options.collation().locale().equals("")) {
+            indexOptions.collation(convert(options.collation()));
```

Comment:
https://github.com/mongodb/morphia/blob/7bd6a89067dcca0d3ecccf99668ec8f3e9da5827/morphia/src/test/java/org/mongodb/morphia/indexes/TestIndexes.java#L65-L65

This test looked for the locale setting.  I went ahead and expanded it to check all the other values as well.